src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -727,6 +736,17 @@ DatabaseShardImp::onChildrenStopped()
 
 void
 DatabaseShardImp::import(Database& source)
+{
+    // Create a temporary TaskQueue to run the lengthy
+    // import process in the background on a dedicated
+    // thread.
+    static TaskQueue importer(*this, "Shard store import");
```

Comment:
This `TaskQueue` will only have one worker thread, and this thread will just be paused forever once the import finishes. And of course this only happens for a node that decides to run an import. So I don't think there's any harm with this approach. I'm trying to avoid adding members to the `DatabaseShardImp` that will just be used for this one-time import process. I'll add a comment explaining the lifetime.

Reply:
@scottschurr Good point, thanks for catching that. The active worker thread after main() exits can cause issues on a shutdown or crash, and prone to future problems. It's also not great having the static live-on after the import completes. I like the `importer_` as a class member solution. 

@undertome `TaskQueue` is overkill for this purpose and I don't see why a thread could not be used instead.

eg.
```
void
DatabaseShardImp::import(Database& source)
{
    // Only the application local node store can be imported
    if (&source != &app_.getNodeStore())
    {
        JLOG(j_.error()) << "invalid source database";
        return;
    }

    std::lock_guard lock(mutex_);

    if (importer_.joinable())
    {
        JLOG(j_.error()) << "Import process already started";
        return;
    }
    importer_ =
        std::thread(&DatabaseShardImp::doImport, this, std::ref(source));
}
```

Then join the thread in ```DatabaseShardImp::onStop()```

```
    if (importer_.joinable())
        importer_.join();
```

Reply:
Done